include/GcInfo/GcInfo.h

```diff
@@ -124,20 +185,21 @@ class GcInfoEmitter {
   void emitGCInfo(const GcFuncInfo *GcFuncInfo);
   void encodeHeader(const GcFuncInfo *GcFuncInfo);
   void encodeTrackedPointers(const GcFuncInfo *GcFuncInfo);
-  void encodePinnedPointers(const GcFuncInfo *GcFuncInfo);
-  void encodeSpecialSlots(const GcFuncInfo *GcFuncInfo);
-  void encodeGcAggregates(const GcFuncInfo *GcFuncInfo);
+  void encodeUntrackedPointers(const GcFuncInfo *GcFuncInfo);
+  void encodeGcAggregate(const llvm::AllocaInst *Alloca,
+                         const AllocaInfo &AllocaInfo);
   void finalizeEncoding();
   void emitEncoding();
 
   bool needsGCInfo(const llvm::Function *F);
   bool needsPointerReporting(const llvm::Function *F);
 
-  bool isTrackedSlot(GcSlotId SlotID);
-  GcSlotId getSlot(int32_t Offset, GcSlotFlags Flags);
-  GcSlotId getTrackedSlot(int32_t Offset);
-  GcSlotId getAggregateSlot(int32_t Offset);
-  GcSlotId getPinnedSlot(int32_t Offset);
+  bool hasSlot(int32_t Offset) { return SlotMap.find(Offset) != SlotMap.end(); }
+  bool isTrackedSlot(const GcSlotId SlotID);
+  GcSlotId getSlot(const int32_t Offset, const GcSlotFlags Flags);
+  GcSlotId getTrackedSlot(const int32_t Offset);
+  GcSlotId getUntrackedSlot(const int32_t Offset, bool IsPinned = false,
+                            bool IsObjectRef = false);
```

Comment:
Ditto in the methods declared here -- for by-value parameters `const` just adds semantic noise....